### PR TITLE
Things that used to check if you were walking now check your movement speed instead

### PR DESCRIPTION
**Pull request by DamianX**
The flasher currently activates if you walk past it, but are dragging someone who has their movement intent set to RUN. With this PR, the flasher now checks the actual speed at which things are moving.
This means diona can run past flashers, and skellingtons can never even walk past them.
This code is horrible but I feel no shame.
I would like someone to Cunningham's law this PR into something mergeable.

:cl:
* tweak: Portable flashers now only activate if you're going too fast, regardless of your run/walk intent. The same goes for water puddles and explosive wet floor signs.

### CODE REVIEW

File: code/game/machinery/flasher.dm [hunk 1]
```diff
@@ -121,7 +121,7 @@ var/list/obj/machinery/flasher/flashers = list()
 
 	if(istype(AM, /mob/living/carbon))
 		var/mob/living/carbon/M = AM
-		if ((M.m_intent != "walk") && (src.anchored))
+		if ((M.glide_size > 2.4615386) && (src.anchored))
```
Comment (Kurfursten) [inline code/game/machinery/flasher.dm]: Nice

**Comment by bathosbathtime** [comment]
But doesn't it make more sense that it has a general speed check rather than the flashers somehow knowing if you are walking rather than running? I like the idea, anyway.

**Comment by DamianX** [comment]
I may have worded the OP terribly. That's what it does, yes. It now checks the actual speed you're going, rather than whether you're walking or running. That magic number I used is just the glide_size of a walking human. I'm not sure how to make it better.

**Comment by Eneocho** [comment]
> skellingtons can never even walk past them

On my way to crawl next to a flasher as a skellington

**Comment by d3athrow** [comment]
do this for other walk/run traps - wet floor slipping and the exploding wet floor sign

**Comment by DamianX** [comment]
it is done

**Comment by DamianX** [comment]
I don't know why that test is failing, butter seems to correctly slip you if you're running

**Comment by calzilla1** [comment]
I'd leave water slips out of it, as walking, in this case, means more "looking where you're going" than the act of actually moving slowly

**Comment by Blithering** [comment]
now we just need to get one of those traffic cop speed guns so sec can pull the scientists zine junkies over for speeding

**Comment by GreatBigFailure** [comment]
Will this affect cloak of cloaking too? I don't want to see the wombo combo of cloak, boots of blinding speed, and scrying orb nerfed

**Comment by Kurfursten** [comment]
Yes, it would break that combo if you end up moving faster than a normal human run

**Comment by Kurfursten** [comment]
What's going on with that butter slipping test?

**Comment by DamianX** [comment]
Like I said, I tested it and it works as intended, except it doesn't in the unit test specifically. I would ping @ShiftyRail but I kinda just want to remove the test

**Comment by Kurfursten** [comment]
Okay, very mysterious. Give shifty a day or two to pop in and have a look at the unit test and if he doesn't have a plan we can just kill the test.

**Comment by ShiftyRail** [comment]
I will try to understand what is going on tommorow, maybe.

**Comment by Kurfursten** [comment]
Gonna give you one more day on this, balancing your schedule against a popular PR

**Comment by ShiftyRail** [comment]
Sorry I am very late at this. It fails because if you run on butter, you do not slip, even though you should. That's why the linter fails.

**Comment by Kurfursten** [comment]
Just wipe that test from the face of the repo and I'll merge it

**Comment by DamianX** [comment]
He implied the test is fine. I'll look at this again today probably

**Comment by Kurfursten** [comment]
I thought he was just being cheeky

**Comment by DamianX** [comment]
Yep, this is the third time I've said this in this PR: the reason the test is failing is because the test dummy doesn't slip on butter if you're running. Except you do. Tested it manually once again: you slip if you run, you don't if you walk.

**Comment by Kurfursten** [comment]
Should be fine to just axe it

**Pull request merged**